Foundation/FileManager.swift

```diff
@@ -618,7 +793,28 @@ open class FileManager : NSObject {
         })
         return contents
     }
-    
+
+#if os(Windows)
+    private func joinPath(prefix: String, suffix: String) -> String {
+      let pszPath: UnsafeMutablePointer<PWSTR?> =
+          UnsafeMutablePointer<PWSTR?>(bitPattern: 0)!
```

Comment:
```suggestion
```

Reply:
Without the explicit initializer, I was getting an error due to the DI failing IIRC

Reply:
If you do need an explicit initialiser to work around a bug, initialising to a non-zero constant should work.

Reply:
I'm worried about that because that gets passed into the Windows system side.  It's a nullable pointer, why is using NULL a bad idea?

Reply:
Using `nil` isn't the issue; the issue is force-unwrapping a `nil` value. If it will let you use `var pszPath: UnsafeMutablePointer<PWSTR?>? = nil` (which I think it should) then go with that.

Reply:
Thinking about it more, I think this should be written:
```
var pszPath: PWSTR? = nil
prefix.withCString(encodedAs: UTF16.self) { prefix in
         suffix.withCString(encodedAs: UTF16.self) { suffix in
           guard PathAllocCombine(prefix, suffix,
                                  ULONG(PATHCCH_ALLOW_LONG_PATHS.rawValue),
                                  &pszPath) == S_OK else { // take the pszPath inout
             return String()
           }
         }
       }
```
instead; I hadn't realised `pszPath` was an out pointer.

Reply:
Ah, derp.  Of course, the inout makes more sense.